scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/synchronization/SynchronizationWrapper.java

```diff
@@ -53,170 +58,221 @@
 
     TaskId taskId;
 
-    SynchronizationInternal internalAPI;
+    transient SynchronizationInternal internalAPI;
+
+    String[] internalAPIUrls;
+
+    static Logger logger = Logger.getLogger(SynchronizationWrapper.class);
 
     public SynchronizationWrapper(String originator, TaskId taskId, SynchronizationInternal internalAPI) {
         this.originator = originator;
         this.taskId = taskId;
         this.internalAPI = internalAPI;
+        if (internalAPI != null && MOP.isReifiedObject(internalAPI)) {
+            internalAPIUrls = PAActiveObject.getUrls(internalAPI);
+        }
+    }
+
+    private void initInternalAPI() {
+        if (this.internalAPI == null && this.internalAPIUrls != null) {
+            for (String lookupURL : this.internalAPIUrls) {
+                try {
+                    this.internalAPI = PAActiveObject.lookupActive(SynchronizationInternal.class, lookupURL);
+                    return;
+                } catch (Exception e) {
```

Comment:
Maybe we can distinguish case when we cannot lookup because it is not alive and other cases.